tonic/src/transport/server.rs

```diff
@@ -421,3 +511,27 @@ where
         })
     }
 }
+
+#[derive(Default, Clone, Debug)]
+#[doc(hidden)]
+pub struct Unimplemented;
+
+impl Service<Request<Body>> for Unimplemented {
+    type Response = Response<BoxBody>;
+    type Error = crate::Error;
+    type Future = future::Ready<Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Ok(()).into()
+    }
+
+    fn call(&mut self, _req: Request<Body>) -> Self::Future {
```

Comment:
do you think this should record a tracing event as well? that way, users get some visibility into when requests hit the unimplemented route (in logs, metrics, etc...)

Reply:
Ummm probably a good idea, but I'd like to add tracing in another PR.